From 4f4246f33e993dc49044ee8430cbb47218844b4c Mon Sep 17 00:00:00 2001
From: Thomas Bourgoin <thomas.bourgoin@foss.st.com>
Date: Mon, 26 Feb 2024 18:09:24 +0100
Subject: [PATCH 0678/1695] plat-stm32mp1: conf: disable CRYP when SAES is
 enable on STM32MP13.

override value of CFG_STM32_CRYP to n when CFG_STM32_SAES=y.
OP-TEE crypto framework can only register one AES accelerator.
If CRYP and SAES are probed, OP-TEE will panic.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git 7d04a94aa77999182b7cceef7b5b290c1b3d453e plat-stm32mp1: conf: disable CRYP when SAES is enable on STM32MP13.]
Signed-off-by: Thomas Bourgoin <thomas.bourgoin@foss.st.com>
Change-Id: Ifccf685eae0c445ff73a13a22029bf193965a467
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/363034
Reviewed-by: Gatien CHEVALLIER <gatien.chevallier@st.com>
Domain-Review: Gatien CHEVALLIER <gatien.chevallier@st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
---
 core/arch/arm/plat-stm32mp1/conf.mk | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/core/arch/arm/plat-stm32mp1/conf.mk b/core/arch/arm/plat-stm32mp1/conf.mk
index f94115512..8f70a2b49 100644
--- a/core/arch/arm/plat-stm32mp1/conf.mk
+++ b/core/arch/arm/plat-stm32mp1/conf.mk
@@ -307,6 +307,14 @@ $(call force,CFG_STM32_I2C,y)
 $(call force,CFG_STM32_GPIO,y)
 endif
 
+#SAES and CRYP cannot be register at the same time in the crypto framework
+#If CRYP and SAES are enable on STMP32MP13, disable CRYP for safety purpose
+ifeq ($(CFG_STM32MP13),y)
+ifeq ($(call cfg-all-enabled, CFG_STM32_CRYP CFG_STM32_SAES), y)
+override CFG_STM32_CRYP := n
+endif # cfg-all-enabled, CFG_STM32_CRYP CFG_STM32_SAES
+endif # CFG_STM32MP13
+
 # If any crypto driver is enabled, enable the crypto-framework layer
 ifeq ($(call cfg-one-enabled, CFG_STM32_CRYP CFG_STM32_SAES),y)
 $(call force,CFG_STM32_CRYPTO_DRIVER,y)
-- 
2.47.2

